webapp/src/tutorialCodeValidation.tsx

```diff
@@ -32,15 +33,36 @@ export class MoveOn extends data.Component<TutorialCodeValidationProps, tutorial
     }
 
     moveOnToNextTutorialStep() {
+        this.validationRuleStatus(false);
         this.props.onYesButtonClick();
         this.showUnusedBlocksMessage(false);
     }
 
     stayOnThisTutorialStep() {
+        this.validationRuleStatus(true);
         this.props.onNoButtonClick();
     }
 
 
+    validationRuleStatus(isEditing: boolean) {
+        const { tutorialName, tutorialStepInfo, tutorialStep } = this.props.options;
+        const stepInfo = tutorialStepInfo[tutorialStep];
+        const rules = stepInfo.listOfValidationRules;
+
+        if (rules != undefined) {
+            const validationRuleStepStatus: pxt.Map<string | number> = {};
+            validationRuleStepStatus["tutorial"] = tutorialName;
+            validationRuleStepStatus["step"] = tutorialStep;
+            for (let i = 0; i < rules.length; i++) {
+                if (rules[i].ruleTurnOn) {
+                    validationRuleStepStatus["ruleName"] = rules[i].ruleName;
+                    let str = 'tutorial.validation' + (isEditing ? '.edit' : '.continue') + (rules[i].ruleStatus ? '.pass' : '.fail');
+                    pxt.tickEvent(str, validationRuleStepStatus);
+                }
+            }
+        }
```

Comment:
Might be worth extracting this to a helper function as this is duplicated from `doubleClickedNextStep` and we tend to have issues consistently updating tick events when they're duplicated like this (e.g. we'll add a property in one place and forget the other)